en/engine/event/event-emit.md

```diff
@@ -0,0 +1,150 @@
+
+Below are what we recommend you to put in:
+
+```ts
+import { _decorator, Component, Node } from "cc";
+const { ccclass } = _decorator;
+
+@ccclass("Example")
+export class Example extends Component {
+    onEnable(){
+        this.node.on('foobar', this._sayHello, this);
+    }
+
+    onDisable(){
+        this.node.off('foobar', this._sayHello, this);
+    }
+
+    _sayHello(){
+        console.log('Hello World');
+    }
+}
+```
+
+## Event Dispatches
+
+There are two ways to trigger an event: `emit` and `dispatchEvent`. The difference between the two is that the latter can do event passing. Let's start with a simple example of the `emit` event.
+
+```ts
+// At most 5 args could be emit.
+xxx.emit(type, ...args);
+```
+
+## Explanation for event arguments
+
+When emitting event, We can start passing our event parameters at the second argument of the emit function. At the same time, the corresponding event parameters are available in the `on` registered callback.
+
+```ts
+import { _decorator, Component, Node } from "cc";
+const { ccclass } = _decorator;
+
+@ccclass("Example")
+export class Example extends Component {
+    onLoad () {
+        this.node.on('foo', (arg1, arg2, arg3) => {
+            console.log(arg1, arg2, arg3);  // print 1, 2, 3
+        });
+    }
+
+    start () {
+        let arg1 = 1, arg2 = 2, arg3 = 3;
+        // At most 5 args could be emit.
+        this.node.emit('foo', arg1, arg2, arg3);
+  }
+}
+```
+
+Note that only up to 5 event parameters can be passed here for the performance of the underlying event distribution. Therefore, care should be taken to control the number of parameters passed when passing a parameter.
+
+## Event delivery
+
+Events launched by the `dispatchEvent` method mentioned above would enter the event delivery stage. In Cocos Creator's event delivery system, we use bubble delivery. Bubble delivery will pass the event from the initiating node continually on to its parent node,  until the root node is reached or an interrupt `event.propagationStopped = true` is made in the response function of a node.
+
+![bubble-event](bubble-event.png)
+
+As shown in the picture above, when we send the event `“foobar”` from node c, if both node a and b listen to the event `“foobar”`, the event will pass to node b and a from c. For example:
+
+```ts
+// In the component script of node c
+this.node.dispatchEvent( new Event.EventCustom('foobar', true) );
+```
+
+If we want to stop the event delivery after node b intercepts the event, we can call the function `event.propagationStopped = true` to do this. Detailed methods are as follows:
+
+```ts
+// In the component script of node b
+this.node.on('foobar', (event: EventCustom) => {
+  event.propagationStopped = true;
+});
+```
+
+Be noted, when you want to dispatch a custom event, please do not use `cc.Event` because it's an abstract class, instead, you should use `cc.Event.EventCustom` to dispatch a custom event.
```

Comment:
Be noted, when you want to dispatch a custom event, please do not use `cc.Event` because it's an abstract class, instead, you should use `cc.Event.EventCustom` to dispatch a custom event.
-->
> **Note**: to dispatch a custom event, do not use `cc.Event` because it's an abstract class. Instead, use `cc.Event.EventCustom` to dispatch a custom event.